src/games/TicTacToe.ts

```diff
@@ -0,0 +1,248 @@
+        ];
+
+        const gameChannel = await message.guild.createChannel(channelName, "text", permissions) as Discord.TextChannel;
+        let gameCategory = message.guild.channels.filter(c => c.type === "category").find(c => c.name === "games");
+        if (!gameCategory) {
+            gameCategory = await message.guild.createChannel("games", "category");
+        }
+        gameChannel.setParent(gameCategory);
+
+        // setup game state
+        const game = {} as TicTacToeGame;
+        this.games.push(game);
+
+        const random = Math.floor(Math.random() * 2);
+        game.Xplayer = random ? message.author.id : opponent.id;
+        game.Oplayer = random ? opponent.id : message.author.id;
+        game.channel = gameChannel;
+        game.turnPlayer = TicTacToeType.O;
+        game.board = {};
+        game.scoreKey = key;
+
+        // print the "welcome" message
+        const status = this.scores[key] || (this.scores[key] = { highWins: 0, lowWins: 0, draws: 0 });
+        const gamesPlayed = status.highWins + status.lowWins + status.draws;
+        const startingPlayer = random ? senderName : opponentName;
+        gameChannel.send(`Game ${gamesPlayed + 1} of ${message.author} vs ${opponent}\nThe gods have decided that ${startingPlayer} goes first`);
+
+        this.incrementGameState(game, -1);
+    }
+
+    private printGameBoard(game: TicTacToeGame) {
+        const tl = game.board[1] || "1";
+        const tm = game.board[2] || "2";
+        const tr = game.board[3] || "3";
+
+        const ml = game.board[4] || "4";
+        const mm = game.board[5] || "5";
+        const mr = game.board[6] || "6";
+
+        const bl = game.board[7] || "7";
+        const bm = game.board[8] || "8";
+        const br = game.board[9] || "9";
+
+        game.channel.send(`\`\`\`${tl} | ${tm} | ${tr}\n---------\n${ml} | ${mm} | ${mr}\n---------\n${bl} | ${bm} | ${br}\`\`\``);
+    }
+
+    private handleChat(message: Discord.Message) {
+        const game = this.games.find(g => g.channel.id === message.channel.id);
+        if (!game) return;
+
+        // ignore messages from the other player
+        if (game.turnPlayer === TicTacToeType.X) {
+            if (message.author.id === game.Oplayer) {
+                return;
+            }
+        }
+        if (game.turnPlayer === TicTacToeType.O) {
+            if (message.author.id === game.Xplayer) {
+                return;
+            }
+        }
+
+        const regex = /\d/;
+        if (regex.test(message.cleanContent)) {
+            const move = +message.cleanContent;
+            if (this.isValidMove(game.board, move)) {
+                this.incrementGameState(game, move);
+            }
+        }
+    }
+
+    private isValidMove(board: { [key: number]: TicTacToeType }, move: number) {
+        return board[move] === undefined;
+    }
+
+    private incrementGameState(game: TicTacToeGame, move: number) {
+        if (move) {
```

Comment:
You call this with -1 in the invite handler. -1 is thruthy so you will set the -1th element of the board, which shouldn't be the intention. move > 0 should be good.